tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/workbench/texteditor/tests/FindReplaceUITest.java

```diff
@@ -0,0 +1,314 @@
+		dialog.setFindText("(a|bc)", false);
+		IFindReplaceTarget target= dialog.getTarget();
+		dialog.simulateEnterInFindInputField(false);
+		assertEquals(0, (target.getSelection()).x);
+		assertEquals(1, (target.getSelection()).y);
+
+		dialog.simulateEnterInFindInputField(false);
+		assertEquals(1, (target.getSelection()).x);
+		assertEquals(2, (target.getSelection()).y);
+	}
+
+	@Test
+	public void testSimpleReplace() {
+		initializeFindReplaceUI("ABCD ABCD ABCD");
+		dialog.setFindText("ABCD", false);
+		dialog.setReplaceText("abcd");
+		dialog.performReplace();
+		runEventQueue();
+
+		assertThat(fTextViewer.getDocument().get(), is("abcd ABCD ABCD"));
+
+		dialog.pressReplaceAll();
+		runEventQueue();
+		assertThat(fTextViewer.getDocument().get(), is("abcd abcd abcd"));
+
+		dialog.select(SearchOptions.REGEX);
+		dialog.setFindText("(ab|cd)", false);
+		dialog.setReplaceText("o");
+		dialog.pressReplaceAll();
+		assertThat(fTextViewer.getDocument().get(), is("oo oo oo"));
+	}
+
+	@Test
+	public void testReplaceAllInSelection() {
+		fTextViewer= new TextViewer(PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell(), SWT.BORDER | SWT.V_SCROLL | SWT.H_SCROLL);
+		fTextViewer.setDocument(new Document("line" + System.lineSeparator() + "line" + System.lineSeparator() + "line"));
+		fTextViewer.getControl().setFocus();
+		fTextViewer.setSelection(new TextSelection(5, 9));
+
+		dialog= openUIFromTextViewer(fTextViewer);
+		assertInitialConfiguration();
+		dialog.unselect(SearchOptions.GLOBAL);
+		dialog.unselect(SearchOptions.WHOLE_WORD);
+		dialog.setFindText("line", false);
+		dialog.setReplaceText("");
+		dialog.pressReplaceAll();
+		runEventQueue();
+		assertThat(fTextViewer.getDocument().get(), is("line" + System.lineSeparator() + System.lineSeparator()));
+	}
+
+
+	@Test
+	public void testChangeInputForIncrementalSearch() {
+		initializeFindReplaceUI("line");
+		dialog.select(SearchOptions.INCREMENTAL);
+
+		dialog.setFindText("lin", false);
+		IFindReplaceTarget target= dialog.getTarget();
+		assertEquals(0, (target.getSelection()).x);
+		assertEquals(dialog.getFindText().length(), (target.getSelection()).y);
+
+		dialog.setFindText("line", false);
+		assertEquals(0, (target.getSelection()).x);
+		assertEquals(dialog.getFindText().length(), (target.getSelection()).y);
+	}
+
+	@Test
+	public void testActivateWholeWordsAndSearchForTwoWords() {
+		initializeFindReplaceUI("text text text");
+		dialog.select(SearchOptions.WHOLE_WORD);
+		dialog.select(SearchOptions.INCREMENTAL);
+
+		dialog.setFindText("text text", false);
+		assertDisabled(SearchOptions.WHOLE_WORD);
+		assertSelected(SearchOptions.WHOLE_WORD);
+
+		dialog.setFindText("text", false);
+		assertEnabled(SearchOptions.WHOLE_WORD);
+		assertSelected(SearchOptions.WHOLE_WORD);
+
+		dialog.select(SearchOptions.REGEX);
+		assertDisabled(SearchOptions.WHOLE_WORD);
+		assertSelected(SearchOptions.WHOLE_WORD);
+	}
+
+
+	public abstract void assertInitialConfiguration();
```

Comment:
This should be part of the `IFindReplaceUIAccess` class, as that class is also responsible for restoring the initial configuration and is thus able to properly evaluate it. This will also allow to assert the initial configuration on closing the UI, as it was the case before this PR.

Reply:
In order to extract this, I have also extracted the following methods into the Access-Interface: 
```java
        void assertUnselected(SearchOptions option);
	void assertSelected(SearchOptions option);
	void assertDisabled(SearchOptions option);
	void assertEnabled(SearchOptions option);
```

Reply:
Moved in https://github.com/eclipse-platform/eclipse.platform.ui/commit/49851d64fd5a67634b90f951a0aeddf3da74bfaa